utils/self-extracting-executable/decompressor.cpp

```diff
@@ -361,6 +361,8 @@ int decompressFiles(int input_fd, char * path, char * name, bool & have_compress
 
 #endif
 
+#if !defined(OS_DARWIN) && !defined(OS_FREEBSD)
```

Comment:
It can be written more naturally with
```
#if defined(OS_LINUX)
```

because other systems like Illumos also don't have it.

There was a port of ClickHouse to Illumos, but it was not added to CI:
https://github.com/ClickHouse/ClickHouse/pulls?q=is%3Apr+illumos+is%3Aclosed

Reply:
I made it consistent with previous distinction for read_exe_path function which also relies on procfs. And BTW it means it will not work for Illumos - this function is not optional. However we don't build distribution for Illumos, so I think currently it's ok.